### ISSUE TITLE
conform.py fails for numpy 2.0.0 and later

### ISSUE THREAD

**Issue by neurolabusc** [Deep-MI/FastSurfer#565]
Attempting to use conform.py with recent versions of numpy (e.g. observed with current stable 2.1.0) creates the error:
```
  AttributeError: `np.obj2sctype` was removed in the NumPy 2.0 release. Use `np.dtype(obj).type` instead.
```

This can be fixed while retaining backward compatibility with two changes:

Change [line 638](https://github.com/Deep-MI/FastSurfer/blob/f2a51b6b125f3cb07b4575191e58040e8939ed8c/FastSurferCNN/data_loader/conform.py#L638) from
```
    sctype = np.uint8 if dtype is None else np.obj2sctype(dtype, default=np.uint8)
```
to read
```
    if np.__version__ < '2.0.0':
        sctype = np.uint8 if dtype is None else np.obj2sctype(dtype, default=np.uint8)
    else:
        sctype = np.uint8 if dtype is None else np.dtype(dtype).type
```

Change [line 798](https://github.com/Deep-MI/FastSurfer/blob/f2a51b6b125f3cb07b4575191e58040e8939ed8c/FastSurferCNN/data_loader/conform.py#L798) from
```
            dtype = np.dtype(np.obj2sctype(dtype)).name
```
to read
```
            if np.__version__ < '2.0.0':
                dtype = np.dtype(np.obj2sctype(dtype)).name
            else:
                dtype = np.dtype(dtype).type.__name__
```

**Comment by m-reuter** [comment]
yes, we do not support numpy 2 yet.

Also would the numpy 2 version (else in your code) be backward compatible or is the if statement really needed?
Thanks.

**Comment by m-reuter** [comment]
just tested it with a numpy array and even in older versions of numpy (1.24.2) can I use the newer statements with the same results as before. So I think this can be simply replaced and no if block is needed.

**Comment by m-reuter** [comment]
Just created a branch and PR #568 to fix this, feel free to try that file. We'll merge this into dev after some quick test and then it should be part of our next release (unless something breaks).

**Comment by m-reuter** [comment]
should be fixed now in dev as #568 is merged. Will also be part of our next release.
